src/collections.rs

```diff
@@ -155,3 +161,149 @@ impl From<&[u8]> for z_bytes_t {
         }
     }
 }
+
+/// A split buffer that owns all of its data.
+///
+/// To minimize copies and reallocations, Zenoh may provide you data in split buffers.
+///
+/// You can use `z_buffer_contiguous` to obtain a contiguous version of a buffer.
+/// If the buffer was already contiguous, the reference count will simply be increased.
+/// Otherwise, the split buffer's entire content will be copied in a newly allocated buffer.
+#[repr(C)]
+pub struct z_owned_buffer_t {
+    _inner: [usize; 5],
+}
+impl_guarded_transmute!(noderefs Option<ZBuf>, z_owned_buffer_t);
+impl Default for z_owned_buffer_t {
+    fn default() -> Self {
+        z_buffer_null()
+    }
+}
+impl From<ZBuf> for z_owned_buffer_t {
+    fn from(value: ZBuf) -> Self {
+        let value = match value.contiguous() {
+            std::borrow::Cow::Borrowed(_) => value,
+            std::borrow::Cow::Owned(value) => value.into(),
+        };
+        unsafe { core::mem::transmute(Some(value)) }
```

Comment:
`core::mem::transmute` should be used ONLY in `impl_guarded_transmute` macro, so no such code should present